plugin/pkg/replacer/replacer_test.go

```diff
@@ -87,46 +117,40 @@ func (m *testHandler) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 }
 
 func TestMetadataReplacement(t *testing.T) {
-
-	mdata := testProvider{
-		"test/key2": func() string { return "two" },
-	}
-
 	tests := []struct {
 		expr   string
 		result string
 	}{
-		{"{name}", "example.org."},
-		{"{/test/key2}", "two"},
-		{"TYPE={type}, NAME={name}, BUFSIZE={>bufsize}, WHAT={/test/key2} .. and more", "TYPE=HINFO, NAME=example.org., BUFSIZE=512, WHAT=two .. and more"},
-		{"{/test/key2}{/test/key4}", "two-"},
-		{"{test/key2", "{test/key2"}, // if last } is missing, the end of format is considered static text
```

Comment:
I guess it worth to keep the corner case of invalid format of the pattern.
=> at least verify that the code support invalid format.

Reply:
yep, let me add a test for that.